src/libostree/ostree-repo-static-delta-compilation-analysis.c

```diff
@@ -217,6 +217,30 @@ string_array_nonempty_intersection (GPtrArray    *a,
   return FALSE;
 }
 
+static gboolean
+sizename_is_delta_candidate (OstreeDeltaContentSizeNames *sizename)
+{
+  /* Don't build candidates for the empty object */
+  if (sizename->size == 0)
+    return FALSE;
+
+  /* Look for known non-delta-able files (currently just compression like xz) */
+  for (guint i = 0; i < sizename->basenames->len; i++)
+    {
+      const char *name = sizename->basenames->pdata[i];
+      /* We could replace this down the line with g_content_type_guess() or so,
+       * but it's not clear to me that's a major win; we'd still need to
+       * maintain a list of compression formats, and this doesn't require
+       * allocation.
+       */
+      if (g_str_has_suffix (name, ".xz"))
```

Comment:
While we're here, let's also catch e.g. `.gz` and `.bzip2` ?

Reply:
`.gz` files can be built with `--rsyncable` which is explicitly designed to be friendly to the rsync-style rollsum algorithm we have.  I don't offhand know if many of them are - we do it for the initramfs which is a key target, but it also doesn't end in `.gz` as a filename. 

It looks like there are *vast* number of these files in atomicws mostly due to man pages, CUPS drivers, Emacs Lisp, and console fonts.  Of those, the only one in Atomic Host is console fonts which rev very infrequently.

Hmm.  Dunno.  I think the problem here is it probably doesn't make sense to try hard to delta most of those, but I'm worried about people who might be using `--rsyncable`.

As far as `.bz2`...there aren't many, I have a grand total of 7 hits in atomicws:

```
find /usr/ -name '*.bz2'
/usr/share/doc/dejavu-fonts-common/status.txt.bz2
/usr/share/doc/guile/NEWS.bz2
/usr/share/doc/libgomp/ChangeLog.bz2
/usr/share/doc/libgomp/ChangeLog.graphite.bz2
/usr/share/doc/ncurses/NEWS.bz2
/usr/share/doc/zfs-fuse/contrib/initrd-for-zfs-fuse-on-root.tar.bz2
/usr/share/selinux/packages/container.pp.bz2
```

All in `/usr/share/doc` except the selinux one.   Yeah, guess we might as well ignore that.

Reply:
> Hmm. Dunno. I think the problem here is it probably doesn't make sense to try hard to delta most of those, but I'm worried about people who might be using `--rsyncable`.

Oh right, we don't want to mess that up. Would be interesting to see if we could find a sweet spot by applying a cut-off file size. Anyway, not worth thinking much more about this for now!